core/job/job.go

```diff
@@ -252,6 +252,20 @@ func (j Jobs) GetJobsWithUnresolvedStaticUpstreams() ([]*WithUpstream, error) {
 	return jobsWithUnresolvedUpstream, me.ToErr()
 }
 
+func (j Jobs) Deduplicate() []*Job {
+	jobByName := map[string]*Job{}
+	for _, subjectJob := range j {
+		jobByName[subjectJob.FullName()] = subjectJob
+	}
+	deduplicatedJobs := make([]*Job, len(jobByName))
+	i := 0
```

Comment:
prefer to minimize the lines
```
for i := 0; i < len(jobByName); i++ {}
```

Reply:
I think it's not possible, as we want to get the value from the map as well